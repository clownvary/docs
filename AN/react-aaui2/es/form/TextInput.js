import _objectWithoutProperties from 'babel-runtime/helpers/objectWithoutProperties';
import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import _extends from 'babel-runtime/helpers/extends';
import React, { PureComponent } from 'react';

import connectForm from './connectForm';
import Input from '../Input';
import { noop, omit } from '../shared/utils';
import { FormFieldAPIPropTypes } from './types';

var propTypes = _extends({}, FormFieldAPIPropTypes);

var TextInput = function (_PureComponent) {
  _inherits(TextInput, _PureComponent);

  function TextInput() {
    var _temp, _this, _ret;

    _classCallCheck(this, TextInput);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, _PureComponent.call.apply(_PureComponent, [this].concat(args))), _this), _this.setWrappedComponentInstance = function (input) {
      _this.input = input;
    }, _this.handleBlur = function (e) {
      var _this$props = _this.props,
          onValidate = _this$props.api.onValidate,
          onBlur = _this$props.onBlur;


      onValidate(_this.input.value);
      onBlur(e);
    }, _this.handleChange = function (e) {
      var _this$props2 = _this.props,
          setValue = _this$props2.api.setValue,
          onChange = _this$props2.onChange;


      setValue(e.target.value);
      onChange(e);
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  TextInput.prototype.render = function render() {
    var rest = _objectWithoutProperties(this.props, []);

    return this.props.static ? React.createElement(
      'span',
      null,
      rest.value
    ) : React.createElement(Input, _extends({
      ref: this.setWrappedComponentInstance,
      type: 'text'
    }, omit(rest, ['api', 'l10n', 'rules']), {
      onBlur: this.handleBlur,
      onChange: this.handleChange
    }));
  };

  return TextInput;
}(PureComponent);

TextInput.displayName = 'TextInput';
TextInput.propTypes = propTypes;
TextInput.defaultProps = {
  onChange: noop,
  onBlur: noop
};


export default connectForm()(TextInput);