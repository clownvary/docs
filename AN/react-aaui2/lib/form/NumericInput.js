'use strict';

exports.__esModule = true;

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _TextInput = require('./TextInput');

var _TextInput2 = _interopRequireDefault(_TextInput);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = { rules: _propTypes.string };
var NumericInput = function NumericInput(props) {
  return _react2.default.createElement(_TextInput2.default, (0, _extends3.default)({ type: 'number' }, props, { rules: props.rules + '|number' }));
};

NumericInput.propTypes = propTypes;
NumericInput.displayName = 'NumericInput';

exports.default = NumericInput;
module.exports = exports['default'];