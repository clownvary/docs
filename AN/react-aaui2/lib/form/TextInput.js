'use strict';

exports.__esModule = true;

var _objectWithoutProperties2 = require('babel-runtime/helpers/objectWithoutProperties');

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _connectForm = require('./connectForm');

var _connectForm2 = _interopRequireDefault(_connectForm);

var _Input = require('../Input');

var _Input2 = _interopRequireDefault(_Input);

var _utils = require('../shared/utils');

var _types = require('./types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var propTypes = (0, _extends3.default)({}, _types.FormFieldAPIPropTypes);

var TextInput = function (_PureComponent) {
  (0, _inherits3.default)(TextInput, _PureComponent);

  function TextInput() {
    var _temp, _this, _ret;

    (0, _classCallCheck3.default)(this, TextInput);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = (0, _possibleConstructorReturn3.default)(this, _PureComponent.call.apply(_PureComponent, [this].concat(args))), _this), _this.setWrappedComponentInstance = function (input) {
      _this.input = input;
    }, _this.handleBlur = function (e) {
      var _this$props = _this.props,
          onValidate = _this$props.api.onValidate,
          onBlur = _this$props.onBlur;


      onValidate(_this.input.value);
      onBlur(e);
    }, _this.handleChange = function (e) {
      var _this$props2 = _this.props,
          setValue = _this$props2.api.setValue,
          onChange = _this$props2.onChange;


      setValue(e.target.value);
      onChange(e);
    }, _temp), (0, _possibleConstructorReturn3.default)(_this, _ret);
  }

  TextInput.prototype.render = function render() {
    var rest = (0, _objectWithoutProperties3.default)(this.props, []);


    return this.props.static ? _react2.default.createElement(
      'span',
      null,
      rest.value
    ) : _react2.default.createElement(_Input2.default, (0, _extends3.default)({
      ref: this.setWrappedComponentInstance,
      type: 'text'
    }, (0, _utils.omit)(rest, ['api', 'l10n', 'rules']), {
      onBlur: this.handleBlur,
      onChange: this.handleChange
    }));
  };

  return TextInput;
}(_react.PureComponent);

TextInput.displayName = 'TextInput';
TextInput.propTypes = propTypes;
TextInput.defaultProps = {
  onChange: _utils.noop,
  onBlur: _utils.noop
};
exports.default = (0, _connectForm2.default)()(TextInput);
module.exports = exports['default'];