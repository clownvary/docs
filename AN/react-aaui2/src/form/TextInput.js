import React, { PureComponent } from 'react'

import connectForm from './connectForm'
import Input from '../Input'
import { noop, omit } from '../shared/utils'
import { FormFieldAPIPropTypes } from './types'

const propTypes = { ...FormFieldAPIPropTypes }

class TextInput extends PureComponent {
  static displayName = 'TextInput'

  static propTypes = propTypes

  static defaultProps = {
    onChange: noop,
    onBlur: noop,
  }

  setWrappedComponentInstance = input => {
    this.input = input
  }

  handleBlur = e => {
    const { api: { onValidate }, onBlur } = this.props

    onValidate(this.input.value)
    onBlur(e)
  }

  handleChange = e => {
    const { api: { setValue }, onChange } = this.props

    setValue(e.target.value)
    onChange(e)
  }

  render() {
    const { ...rest } = this.props

    return this.props.static ? (
      <span>{rest.value}</span>
    ) : (
      <Input
        ref={this.setWrappedComponentInstance}
        type="text"
        {...omit(rest, ['api', 'l10n', 'rules'])}
        onBlur={this.handleBlur}
        onChange={this.handleChange}
      />
    )
  }
}

export default connectForm()(TextInput)
