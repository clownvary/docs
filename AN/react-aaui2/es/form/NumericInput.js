import _extends from 'babel-runtime/helpers/extends';
import React from 'react';
import { string } from 'prop-types';

import TextInput from './TextInput';

var propTypes = { rules: string };
var NumericInput = function NumericInput(props) {
  return React.createElement(TextInput, _extends({ type: 'number' }, props, { rules: props.rules + '|number' }));
};

NumericInput.propTypes = propTypes;
NumericInput.displayName = 'NumericInput';

export default NumericInput;