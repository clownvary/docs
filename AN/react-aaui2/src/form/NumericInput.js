import React from 'react'
import { string } from 'prop-types'

import TextInput from './TextInput'

const propTypes = { rules: string }
const NumericInput = props => (
  <TextInput type="number" {...props} rules={`${props.rules}|number`} />
)

NumericInput.propTypes = propTypes
NumericInput.displayName = 'NumericInput'

export default NumericInput
