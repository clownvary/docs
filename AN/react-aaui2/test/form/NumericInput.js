import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import NumericInput from 'form/NumericInput'
import TextInput from 'form/TextInput'
import formMountOptions from '../__mocks__/formMountOptions'

it('should render without errors', () => {
  const wrapper = mount(<NumericInput />, formMountOptions)

  expect(wrapper).toBeTruthy()
})

it('should render correct things', () => {
  const wrapper = mount(
    <NumericInput name="number" type="number" />,
    formMountOptions,
  )

  expect(wrapper.find('input').length === 1).toBe(true)
  expect(wrapper.prop('name') === 'number').toBe(true)
  expect(wrapper.prop('type') === 'number').toBe(true)
})

it("should render right thing and don't change unexpected", () => {
  const tree = renderer
    .create(
      <NumericInput
        name="number"
        type="number"
        aauiFormStore={formMountOptions.context.aauiFormStore}
      />,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('should has `number` rule by default', () => {
  const wrapper = mount(
    <NumericInput name="number" type="number" />,
    formMountOptions,
  )

  expect(
    wrapper
      .find(TextInput)
      .prop('rules')
      .indexOf('number') !== -1,
  ).toBe(true)
})
