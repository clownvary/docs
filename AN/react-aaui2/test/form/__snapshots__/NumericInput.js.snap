// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `
<div
  className="input-group input-group--m"
  style={undefined}
>
  <input
    aauiFormStore={
      Store {
        "asyncReducers": Object {
          "number": [Function],
          "undefined": [Function],
        },
        "childStoreList": Array [],
        "currentState": Object {
          "number": Object {
            "value": undefined,
          },
          "undefined": Object {
            "value": undefined,
          },
        },
        "getError": [Function],
        "getValue": [Function],
        "initialState": Object {},
        "listeners": Object {
          "clear": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "setError": [Function],
        "setValue": [Function],
        "validators": Object {
          "number": [Function],
          "undefined": [Function],
        },
      }
    }
    className="input input-group__field"
    disabled={undefined}
    name="number"
    onBlur={[Function]}
    onChange={[Function]}
    onInput={[Function]}
    type="number"
  />
</div>
`;
